packages/playground/build.ts

```diff
@@ -1,20 +1,249 @@
+      build.onLoad({ filter: /.*/, namespace: "mml" }, async (args) => {
+        log("onLoad", args);
+        const { path } = args;
+        const result = await build.esbuild.build({
+          ...build.initialOptions,
+          metafile: true,
+          entryPoints: [path],
+          plugins: [makePlugin()]
+        });
+        const outPath = Object.keys(result.metafile.outputs)[0];
+        const relativeOutPath = relative(outdir, outPath).replace(/\.[tj]sx?/, ".html");
+        const importStub = `mml:${relativeOutPath}`;
+
+        importStubs[relativeOutPath] = importStub;
+
+        return { contents: importStub, loader: "text" };
+      });
+
+
+      // Any raw HTML files should just be copied to the build directory.
+      // TODO: These could contain script tags with references to local files,
+      //       we may want to consider loading and embedding them directly into the HTML.
+      build.onLoad({ filter: /\.html$/ }, async (args) => {
+        log("onLoad", args);
+        const { path } = args;
+        const contents = await fsp.readFile(path, { encoding: "utf8" });
+        return { contents, loader: "copy" };
+      });
+
+      build.onEnd(async result => {
+        if (!isRoot) {
+          results.push(result);
+          return;
+        }
+
+        // We are in the root plugin instance. All child instances have finished and
+        // pushed their results into the array. Now we combine all the results into one.
+        const combinedResults = results.reduce((acc, val) => ({
+          errors: acc.errors.concat(val.errors),
+          warnings: acc.warnings.concat(val.warnings),
+          outputFiles: (acc.outputFiles ?? []).concat(val.outputFiles ?? []),
+          metafile: {
+            inputs: { ...acc.metafile?.inputs, ...val.metafile?.inputs },
+            outputs: { ...acc.metafile?.outputs, ...val.metafile?.outputs },
+          },
+          mangleCache: { ...acc.mangleCache, ...val.mangleCache },
+        }), result);
+
+        Object.assign(result, combinedResults);
+
+        if (result.errors.length > 0) {
+          log("onEnd: errors in build, releasing lock on build directory");
+          lockfile.unlockSync(outdir);
+          return;
+        }
+
+        // If we have a any js files, that do not have a corresponding HTML file,
+        // we need to create one and embed the JavaScript into a <script> tag.
+        // Then we can delete the JavaScript file as it is no longer needed.
+        const outputs = result.metafile!.outputs;
+        for (const [jsPath, meta] of Object.entries(outputs)) {
+          if (jsExt.test(jsPath)) {
+            const htmlPath = jsPath.replace(/\.js$/, ".html");
+            if (!(htmlPath in outputs)) {
+              delete outputs[jsPath]
+              const js = await fsp.readFile(jsPath, { encoding: "utf8" });
+              const html = `<body></body><script>${js}</script>`;
```

Comment:
We should probably brush off a few more bytes by removing the multi-line comments that the React build leaves in the built file
```
const cleanedJs = js.replace(/\/\*[\s\S]*?\*\//g, "");
const html = `<body></body><script>${cleanedJs}</script>`;
```